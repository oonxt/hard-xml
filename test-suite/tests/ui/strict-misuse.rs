use hard_xml::XmlRead;

// TODO this compiles. Should it fail instead?
#[derive(Clone, Copy, XmlRead)]
#[xml(strict, tag = "Bare")]
struct Bare {}

// TODO this compiles. Should it fail instead?
#[derive(Clone, Copy, XmlRead)]
#[xml(strict = "all", tag = "KeyValue")]
struct KeyValue {}

#[derive(Clone, Copy, XmlRead)]
#[xml(strict(data), tag = "UnsupportedArgument")]
struct UnsupportedArgument {}

#[derive(Clone, Copy, XmlRead)]
#[xml(strict(data = "unknown_attributes"), tag = "UnsupportedMeta")]
struct UnsupportedMeta {}

fn main() {}
