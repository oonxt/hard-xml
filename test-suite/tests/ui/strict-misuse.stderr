error: unsupported argument to `strict`
  --> tests/ui/strict-misuse.rs:14:14
   |
14 | #[xml(strict(data), tag = "UnsupportedArgument")]
   |              ^^^^

error: unsupported meta type in `strict`
  --> tests/ui/strict-misuse.rs:18:14
   |
18 | #[xml(strict(data = "unknown_attributes"), tag = "UnsupportedMeta")]
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
